### PR TITLE
PRIV-8214 added new public method for registration new child

### PR DESCRIPTION
**Pull request by yoklmnprst**
Registration method - "by parent email".

### CODE REVIEW

File: Sources/PrivoSDK/PrivoConfiguration.swift [hunk 1]
```diff
@@ -16,6 +16,8 @@ public class PrivoConfiguration {
     let ageGatePublicUrl: URL
     let ageVerificationBaseUrl: URL
     let ageVerificationPublicUrl: URL
+    let svcUrl: URL
```
Comment (AlexSlo-Dev) [inline Sources/PrivoSDK/PrivoConfiguration.swift:19]: what SVC? we have more than 30 different SVCs.

File: Sources/PrivoSDK/api/Rest.swift [hunk 2]
```diff
@@ -48,6 +52,8 @@ protocol Restable {
     func processRecheck(data: RecheckStatusRecord) async throws -> AgeGateActionResponse
     func trackCustomError(_ errorDescr: String)
     func sendAnalyticEvent(_ event: AnalyticEvent)
+    func registerParentAndChild(_ parentChildPair: ParentChildPair, _ token: String) async throws -> RegisterResponse
+    func getP3Token(_ clientId: String, _ clientSecret: String) async throws -> TokenResponse
```
Comment (AlexSlo-Dev) [inline Sources/PrivoSDK/api/Rest.swift:56]: 1) why getP3Token is part of URL extension? Do not all urs will be able to call getP3Token
2) getP3Token looks like a strange naming, let's try to rename it

File: Sources/PrivoSDK/model/OAuthToken.swift [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+struct TokenResponse: Decodable {
+    let access_token: String
+    let token_type: String
+    let expires_in: Int
+    let scope: String
+}
+
+extension OAuthToken {
+    func toQueryItems() -> [URLQueryItem] {
+        let mirror = Mirror(reflecting: self)
```
Comment (AlexSlo-Dev) [inline Sources/PrivoSDK/model/OAuthToken.swift:19]: OAuth token do not related to URL or to QueryItems
it means that this converter should be in another place

File: Sources/PrivoSDK/model/RegisterResponse.swift [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import Foundation
+
+struct RegisterResponse: Codable {
+    let to: To
+
+    struct To: Codable {
```
Comment (AlexSlo-Dev) [inline Sources/PrivoSDK/model/RegisterResponse.swift:6]: why do we need an embedded struct here?

File: Sources/PrivoSDK/model/OAuthToken.swift (hunk 3 above)
Comment (yoklmnprst) [reply Sources/PrivoSDK/model/OAuthToken.swift]: after switching request to gw token this code removed as unused

File: Sources/PrivoSDK/model/RegisterResponse.swift (hunk 4 above)
Comment (yoklmnprst) [reply Sources/PrivoSDK/model/RegisterResponse.swift:6]: this allows to use auto-synthesized compiler tricks for Codable and have less and clearer code than if were overriding writing parsing manually

File: Sources/PrivoSDK/api/Rest.swift (hunk 2 above)
Comment (yoklmnprst) [reply Sources/PrivoSDK/api/Rest.swift]: * Discussed and decided to change request to gw token *

File: Sources/PrivoSDK/PrivoConfiguration.swift [hunk 5]
```diff
@@ -31,6 +32,7 @@ public class PrivoConfiguration {
                 ageGatePublicUrl: URL? = nil,
                 ageVerificationBaseUrl: URL? = nil,
                 ageVerificationPublicUrl: URL? = nil,
+                svcUrl: URL? = nil,
```
Comment (AlexSlo-Dev) [inline Sources/PrivoSDK/PrivoConfiguration.swift:35]: why SVC again ?

File: Sources/PrivoSDK/api/Rest.swift [hunk 6]
```diff
@@ -14,12 +14,16 @@ struct URLComponentConstants: ExpressibleByStringLiteral, RawRepresentable {
 }
 
 extension URLComponentConstants {
-    static let analytic: URLComponentConstants = "metrics"
```
Comment (AlexSlo-Dev) [inline Sources/PrivoSDK/api/Rest.swift:17]: why do we need this enum? It's just string path's

File: Sources/PrivoSDK/model/ParentChildPair.swift [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+        let roleIdentifier: String
+        let sendCongratulationsEmail: Bool
+        let attributes: [Attribute]
+        
+        init(child: Child) {
+            self.firstName = child.firstname
+            self.lastName = child.lastname
+            self.userName = child.username
+            let dateFormatter = DateFormatter()
+            dateFormatter.dateFormat = "yyyyMMdd"
```
Comment (AlexSlo-Dev) [inline Sources/PrivoSDK/model/ParentChildPair.swift:28]: dateFormat = "yyyyMMdd" it's cold birthDatePrecision 
please add it to struct as a separate field. See example here - https://consent-dev.privo.com/#/demo

File: Sources/PrivoSDK/PrivoConfiguration.swift (hunk 5 above)
Comment (yoklmnprst) [reply Sources/PrivoSDK/PrivoConfiguration.swift:35]: fixed

File: Sources/PrivoSDK/api/Rest.swift (hunk 6 above)
Comment (yoklmnprst) [reply Sources/PrivoSDK/api/Rest.swift:17]: updated

File: Sources/PrivoSDK/PrivoAuth.swift [hunk 8]
```diff
@@ -217,4 +217,20 @@ public class PrivoAuth {
...
+        
+        let parentChildPair = ParentChildPair(
+            roleIdentifier: RoleIdentifier.parentStandard.rawValue,
+            email: parentEmail,
+            minorRegistrations: [
+                .init(child: child)
+            ])
+        let response = try await api.registerParentAndChild(parentChildPair, gwToken)
+        
+        let updatePasswordLink = URL(string: response.to.updatePasswordLink) ?? URL(fileURLWithPath: "")
```
[the remark this answers is not in the document]
Comment (yoklmnprst) [reply Sources/PrivoSDK/PrivoAuth.swift:233]: fixed this issue

File: Sources/PrivoSDK/model/ParentChildPair.swift (hunk 7 above)
Comment (yoklmnprst) [reply Sources/PrivoSDK/model/ParentChildPair.swift:28]: implemented

**Review by AlexSlo-Dev — APPROVED** [review]

**Pull request merged**